.github/workflows/compilation_on_android_ubuntu.yml

```diff
@@ -493,6 +497,13 @@ jobs:
       - name: checkout
         uses: actions/checkout@v3
 
+      - name: download and install wasi-sdk
+        run: |
```

Comment:
This step and below step "build socket api tests" will be executed in each running mode and each test_option, which may take a lot of time to run the whole CI. Shall we add the test for some situations, for example, $WASI_TEST_OPTIONS test_option for all running modes? Refer to #1963, here and below, add condition:
```yml
if: matrix.test_option == '$WASI_TEST_OPTIONS'
```